snowflakeid/src/main/java/snowflakeid/SnowflakeIdGenerator.java

```diff
@@ -0,0 +1,75 @@
+package snowflakeid;
+
+import java.time.Instant;
+
+public class SnowflakeIdGenerator {
+
+    private final long defaultEpoch = 1759536000000L;
+    private final long machineIdBits = 10L;
+    private final long maxMachineId = ~(-1L << machineIdBits);
+    private final long sequenceBits = 12L;
+
+    private long machineId;
+
+    public long getSequence() {
+        return sequence;
+    }
+
+    public void setSequence(long sequence) {
+        this.sequence = sequence;
+    }
+
+    private long sequence = -1L;
+
+    private long lastTimestamp = -1L;
+
+    public SnowflakeIdGenerator(long machineId) {
```

Comment:
Very error-prone, if the consumer accidentally passes in the same machineId for different threads, this will break the specs.